python-package/lightgbm/plotting.py

```diff
@@ -513,7 +515,13 @@ def add(
                         missing_type_str=root['missing_type'],
                         default_left=root['default_left']
                     )
-            label += f"<B>{_float2str(root['threshold'], precision)}</B>"
+            if max_category_values and root['decision_type'] == '==' and len(root['threshold'].split('||')) > max_category_values:
```

Comment:
Can you make this check specifically for `None`, i.e. `if max_category_values is not None`? and maybe move the other two conditions to that inner if? I think if you're going through the code and you're not interested in the path that `max_category_values` takes its easier to see that single if and go to the else or skip it entirely.

Reply:
I have defaulted ```threshold = root['threshold'] ``` right after setting ```tooltip = None``` and now the if condition looks better
```diff
@@ -479,6 +479,7 @@ def _to_graphviz(
         fillcolor = 'white'
         style = ''
         tooltip = None
+        threshold = root['threshold']
         if highlight:
             color = 'blue'
             penwidth = '3'
@@ -515,11 +516,10 @@ def _to_graphviz(
                         missing_type_str=root['missing_type'],
                         default_left=root['default_left']
                     )
-            if max_category_values and root['decision_type'] == '==' and len(root['threshold'].split('||')) > max_category_values:
-                tooltip = root['threshold']
-                threshold = '...'
-            else:
-                threshold = root['threshold']
+            if max_category_values is not None:
+                if root['decision_type'] == '==' and len(root['threshold'].split('||')) > max_category_values:
+                    tooltip = root['threshold']
+                    threshold = '...'

```

Reply:
I think this caused a problem with the automated test, the line
```diff
+        threshold = root['threshold']
```

should be under the condition
```python
if 'split_index' in root:  # non-leaf
```
I will test it and commit it